src/helpers.js

```diff
@@ -8,17 +11,17 @@ export const isElementAnSFC = (element) => {
   return !element.type.prototype.isReactComponent;
 };
 
-export function omit(obj, attrs = []) {
-  const result = {};
-  Object.keys(obj).forEach((key) => {
+export function omit<R: {}, T: R>(obj: T, attrs: $Keys<T>[] = []): R {
+  const result: $Shape<T> = {};
+  Object.keys(obj).forEach((key: $Keys<T>) => {
```

Comment:
For fun, a friend and I were trying to see if we could come up with a way within Flow to specify `omit`'s behaviour, using `$Diff` and `$Keys`. We couldn't quite get there, alas, but my friend did come up with what I think is a simpler implementation:

```js
export function omit<T: {}>(obj: T, attrs: Array<$Keys<T>> = []): $Shape<T> {
  const result = {};
  Object.keys(obj).forEach((key: $Keys<T>) => {
    if (attrs.indexOf(key) === -1) {
      result[key] = obj[key];
    }
  });
  return result;
}
``` 

Let me know if there's a reason you need the second generic.